### PR TITLE
Attempt to implement tempbanning.

### PR DESCRIPTION
**Pull request by Jayderp**
Making a pull request  to implement tempbanning.
I know this will NOT work, at all, but I want benksu (or whoever is the main dev of skript idk pls no ddos) to implement this. ok thanks :D
What this will do (if coded correctly):
takes a timespan (if any), adds it to the current time, then bans the targeted player, for the specified reason (if none, ban for default), and for the specified time (if specified).

ok thanks pls look at it <3

### Description

---
**Target Minecraft Versions:**     any
**Requirements:**     any
**Related Issues:**     #2667

### CODE REVIEW

File: src/main/java/ch/njol/skript/effects/EffBan.java [hunk 1]
```diff
@@ -73,6 +73,7 @@
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
 		players = exprs[0];
 		reason = exprs.length > 1 ? (Expression<String>) exprs[1] : null;
+		duration = exprs.length > 2 ? (Expression<String>) exprs[2] : null;
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/effects/EffBan.java]: Since this is a Timespan, it should be ``Expression<Timespan>`` and not ``Expression<String>``

Comment (Wealthyturtle) [reply src/main/java/ch/njol/skript/effects/EffBan.java]: You are also missing the declaration of the ``duration`` variable, please append this to your code:
```
@Nullable
private Expression<Timespan> duration;
```

Comment (Wealthyturtle) [reply src/main/java/ch/njol/skript/effects/EffBan.java]: Remember to import the Timestamp class also, via ```import ch.njol.skript.util.Timespan;```

File: src/main/java/ch/njol/skript/effects/EffBan.java [hunk 2]
```diff
@@ -81,7 +82,7 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	@Override
 	protected void execute(final Event e) {
 		final String reason = this.reason != null ? this.reason.getSingle(e) : null; // don't check for null, just ignore an invalid reason
-		final Date expires = null;
+		final Date expires = duration; // pseudo code
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/effects/EffBan.java]: ``duration`` is a ``Timespan`` object, while ``expires`` is a ``Date`` object. Instead, use:
```
final Date expires = new Date(System.currentTimeMillis() + duration.getMilliSeconds());
```
*I think this is the right way to create the Date object, haven't worked with Date objects before... if it's wrong someone please correct me lol*

**Comment by Jayderp** [comment]
sha-sha-sha-sha-SHANEBEE will probably improve this

File: src/main/java/ch/njol/skript/effects/EffBan.java [hunk 3]
```diff
@@ -55,15 +57,16 @@
...
+				"ban %strings/offlineplayers% [(by reason of|because [of]|on account of|due to) %-string%] [for %timespan%]", "unban %strings/offlineplayers%",
+				"ban %players% by IP [(by reason of|because [of]|on account of|due to) %-string%] [for %timespan%]", "unban %players% by IP",
+				"IP(-| )ban %players% [(by reason of|because [of]|on account of|due to) %-string%] [for %timespan%]", "(IP(-| )unban|un[-]IP[-]ban) %players%");
 	}
 	
 	@SuppressWarnings("null")
 	private Expression<?> players;
 	@Nullable
 	private Expression<String> reason;
+	private Expression<Timespan> duration;
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/effects/EffBan.java]: Missing the ``@Nullable`` tag above it

File: src/main/java/ch/njol/skript/effects/EffBan.java [hunk 4]
```diff
@@ -81,7 +85,7 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	@Override
 	protected void execute(final Event e) {
 		final String reason = this.reason != null ? this.reason.getSingle(e) : null; // don't check for null, just ignore an invalid reason
-		final Date expires = null;
+		final Date expires = new Date(System.currentTimeMillis() + duration.getMilliSeconds());
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/effects/EffBan.java]: Oops, my bad should have been
```
final Date expires = new Date(System.currentTimeMillis() + this.duration.getSingle(e).getMilliSeconds());
```

File: src/main/java/ch/njol/skript/effects/EffBan.java (hunk 3 above)
Comment (Jayderp) [reply src/main/java/ch/njol/skript/effects/EffBan.java]: oops thought i could combine those lol

File: src/main/java/ch/njol/skript/effects/EffBan.java [hunk 5]
```diff
@@ -81,7 +88,7 @@ public boolean init(final Expression<?>[] exprs, final int matchedPattern, final
 	@Override
 	protected void execute(final Event e) {
 		final String reason = this.reason != null ? this.reason.getSingle(e) : null; // don't check for null, just ignore an invalid reason
-		final Date expires = null;
+		final Date expires = new Date(System.currentTimeMillis() + this.duration.getSingle(e).getMilliSeconds());
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/effects/EffBan.java]: Third time's the charm... use
```
final Date expires = this.duration != null ? new Date(System.currentTimeMillis() + this.duration.getSingle(e).getMilliSeconds()) : null;
```

**Comment by bensku** [comment]
If you want us to implement something, don't open a (broken) pull request for it.

**Pull request closed without merging**